lib/mina/kubernetes.rb

```diff
@@ -85,16 +84,37 @@ def image_available?(commit)
 
 def run_command(command, env_hash = {})
   env = env_hash.collect{|k,v| "--env #{k}=#{v}" }.join(" ")
-  label = command.downcase.gsub(" ", "-").gsub(":", "-")
+  label = command.downcase.gsub(" ", "-").gsub(":", "-")+ "-#{`whoami`}".strip + "-#{fetch(:branch)}" 
   proxy_env = "HTTPS_PROXY=#{fetch(:proxy)}" if fetch(:proxy)
 
-  # using system instead of mina's command so tty opens successfully
-  system "#{proxy_env} kubectl run #{label}-#{SecureRandom.hex(4)} --rm -i --tty --restart=Never --context=#{fetch(:kubernetes_context)} --namespace=#{fetch(:namespace)} --image #{fetch(:image_repo)}:#{fetch(:image_tag)} #{env} -- #{command}"
+  run :local do
+    comment "Lauching Pod #{color(label, 36)} to run #{color(command, 36)}"
+  end
+
+  pod_description = `#{proxy_env} kubectl get pod #{label} -o json --ignore-not-found --context=#{fetch(:kubernetes_context)} --namespace=#{fetch(:namespace)}`
+
+  if pod_description.empty?
+    wait_until_image_ready(fetch(:image_tag))
+    run_command = "#{proxy_env} kubectl run #{label} --rm -i --tty --restart=Never --context=#{fetch(:kubernetes_context)} --namespace=#{fetch(:namespace)} --image #{fetch(:image_repo)}:#{fetch(:image_tag)} #{env}"
+    system "#{run_command} -- #{command}"
+  else
+    started_at = Time.parse(JSON.parse(pod_description)["status"]["startTime"]).strftime('%b %e, %H:%M')
+    choice = TTY::Prompt.new.select("Pod already exists, running since #{started_at} UTC, what would you like to do?", {"Reattach session" => 1, "Kill it" => 0})
+    
+    attach_command = "#{proxy_env} kubectl attach #{label} -i --tty -c #{label} --context=#{fetch(:kubernetes_context)} --namespace=#{fetch(:namespace)}"
+    delete_command = "#{proxy_env} kubectl delete pod #{label} --context=#{fetch(:kubernetes_context)} --namespace=#{fetch(:namespace)}"
+    
+    if choice == 1
+      system "#{attach_command} && #{delete_command}"
```

Comment:
The `--rm` from the original `kubectl run` doesn't seem to apply when reattaching and the pod doesn't get deleted after the session is closed, so had to add an explicit `kubectl delete`